examples/rest-spot-exchange-info.ts

```diff
@@ -0,0 +1,212 @@
+  if (!stepSize) return quantity.toString();
+
+  // Calculate precision from step size
+  const precision = stepSize.indexOf('1') - 1;
+  if (precision < 0) return Math.floor(quantity).toString();
+
+  // Round to the nearest step
+  const stepSizeNum = parseFloat(stepSize);
+  const rounded = Math.floor(quantity / stepSizeNum) * stepSizeNum;
+
+  // Format with correct precision
+  return rounded.toFixed(precision);
+}
+
+/**
+ * Validates and formats an order based on symbol constraints
+ */
+function formatOrderParams(
+  symbol: string,
+  price: number,
+  quantity: number,
+  symbolInfo: any,
+): { symbol: string; price: string; quantity: string } | { error: string } {
+  try {
+    // Check if price is within allowed range
+    const minPrice = parseFloat(symbolInfo.tickSize || '0');
+    if (price < minPrice) {
+      throw new Error(`Price ${price} is below minimum ${minPrice}`);
+    }
+
+    // Check if quantity is within allowed range
+    const minQty = parseFloat(symbolInfo.minOrderQty || '0');
+    const maxQty = parseFloat(symbolInfo.maxOrderQty || Infinity);
+
+    if (quantity < minQty) {
+      throw new Error(`Quantity ${quantity} is below minimum ${minQty}`);
+    }
+
+    if (quantity > maxQty) {
+      throw new Error(`Quantity ${quantity} exceeds maximum ${maxQty}`);
+    }
+
+    // Check notional value (price * quantity)
+    const notional = price * quantity;
+    const minNotional = parseFloat(symbolInfo.minNotional || '0');
+
+    if (notional < minNotional) {
+      throw new Error(
+        `Order value ${notional} is below minimum ${minNotional}`,
+      );
+    }
+
+    // Format price and quantity according to exchange requirements
+    const formattedPrice = roundToTickSize(price, symbolInfo.tickSize);
+    const formattedQty = roundToStepSize(quantity, symbolInfo.qtyStepSize);
+
+    return {
+      symbol,
+      price: formattedPrice,
+      quantity: formattedQty,
+    };
+  } catch (error) {
+    throw new Error(`Failed to format order: ${error.message}`);
+  }
+}
+
+// Example usage
+async function testSymbolUtils() {
+  const info = await exchangeInfo;
+  if (!info) return;
+
+  const symbolFilters = await getSymbolInfo(info, symbol);
+  if (!symbolFilters) return;
+
+  // Test price formatting
+  const testPrice = 23.45678;
+  console.log(`Original price: ${testPrice}`);
+  console.log(
+    `Formatted price: ${roundToTickSize(testPrice, symbolFilters.tickSize.toString())}`,
+  );
+
+  // Test quantity formatting
+  const testQty = 1.23456;
+  console.log(`Original quantity: ${testQty}`);
+  console.log(
+    `Formatted quantity: ${roundToStepSize(
+      testQty,
+      symbolFilters.qtyStepSize.toString(),
+    )}`,
+  );
+
+  // Test full order formatting
+  const orderParams = formatOrderParams(
+    symbol,
+    testPrice,
+    testQty,
+    symbolFilters,
+  );
+  console.log('Formatted order parameters:');
+  console.log(orderParams);
```

Comment:
Nice example! would be good to show a final call to submit order, with the correctly formatted values